src/source/raster_dem_tile_source.js

```diff
@@ -0,0 +1,128 @@
+
+        tile.neighboringTiles = this._getNeighboringTiles(tile.coord.id);
+        function imageLoaded(err, img) {
+            delete tile.request;
+            if (tile.aborted) {
+                this.state = 'unloaded';
+                callback(null);
+            } else if (err) {
+                this.state = 'errored';
+                callback(err);
+            } else if (img) {
+                if (this.map._refreshExpiredTiles) tile.setExpiryData(img);
+                delete (img: any).cacheControl;
+                delete (img: any).expires;
+
+                const rawImageData = browser.getImageData(img);
+                const params = {
+                    uid: tile.uid,
+                    coord: tile.coord,
+                    source: this.id,
+                    rawImageData: rawImageData
+                };
+
+                if (!tile.workerID || tile.state === 'expired') {
+                    tile.workerID = this.dispatcher.send('loadDEMTile', params, done.bind(this));
+                }
+            }
+        }
+
+        function done(err, data) {
+            if (err) {
+                this.state = 'errored';
+                callback(err);
+            }
+
+            if (data) {
+                tile.dem =  DEMData.deserialize(data);
+                tile.state = 'loaded';
+                callback(null);
+            }
+        }
+    }
+
+    _getNeighboringTiles(tileId: number) {
+        const {z, x, y, w} = TileCoord.fromID(tileId);
+        const dim = Math.pow(2, z);
+
+        const px = (x - 1 + dim) % dim;
+        const pxw = x === 0 ? w - 1 : w;
+        const nx = (x + 1 + dim) % dim;
+        const nxw = x + 1 === dim ? w + 1 : w;
+
+        const neighboringTiles = {};
+        neighboringTiles[getTileId({ z: z, x: px, y: y, w: pxw })] = {backfilled: false};
+        neighboringTiles[getTileId({ z: z, x: nx, y: y, w: nxw })] = {backfilled: false};
+
+        // Add upper neighboringTiles
+        if (y > 0) {
+            neighboringTiles[getTileId({ z: z, x: px, y: y - 1, w: pxw  })] = {backfilled: false};
+            neighboringTiles[getTileId({ z: z, x: x, y: y - 1, w: w  })] = {backfilled: false};
+            neighboringTiles[getTileId({ z: z, x: nx, y: y - 1, w: nxw  })] = {backfilled: false};
+        }
+        // Add lower neighboringTiles
+        if (y + 1 < dim) {
+            neighboringTiles[getTileId({ z: z, x: px, y: y + 1, w: pxw  })] = {backfilled: false};
+            neighboringTiles[getTileId({ z: z, x: x, y: y + 1, w: w  })] = {backfilled: false};
+            neighboringTiles[getTileId({ z: z, x: nx, y: y + 1, w: nxw  })] = {backfilled: false};
+        }
+
+        return neighboringTiles;
+
+        function getTileId(coord) {
+            const tilecoord = new TileCoord(coord.z, coord.x, coord.y, coord.w);
+            return tilecoord.id;
+        }
```

Comment:
^ instead of constructing all these new objects, let's extract the TileCoord constructor code that calculates the id from `(z,x,y,w)` as a static method so we can just do `neighboringTiles[TileCoord.idFromCoordinates(z, px, y - 1, pxw)]`